pkg/api/server/v1alpha2/log/gcs.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package log
+
+import (
+	"path/filepath"
+
+	"golang.org/x/oauth2/google"
+
+	"context"
+	"fmt"
+	"io"
+
+	server "github.com/tektoncd/results/pkg/api/server/config"
+	"github.com/tektoncd/results/pkg/apis/v1alpha2"
+
+	"gocloud.dev/blob/gcsblob"
+	"gocloud.dev/gcp"
+)
+
+type gcsStream struct {
+	ctx    context.Context
+	config *server.Config
+	key    string
+	client *gcp.HTTPClient
+}
+
+// NewGCSStream returns a log streamer for the GCS storage type.
+func NewGCSStream(ctx context.Context, log *v1alpha2.Log, config *server.Config) (Stream, error) {
+	if log.Status.Path == "" {
+		filePath, err := FilePath(log)
+		if err != nil {
+			return nil, err
+		}
+		log.Status.Path = filePath
+	}
+
+	filePath := filepath.Join(config.LOGS_PATH, log.Status.Path)
+
+	client, err := getGCSClient(ctx, config)
+	if err != nil {
+		return nil, err
+	}
+
+	gcs := &gcsStream{
+		ctx:    ctx,
+		config: config,
+		key:    filePath,
+		client: client,
+	}
+
+	return gcs, nil
+}
+
+func getGCSClient(ctx context.Context, cfg *server.Config) (*gcp.HTTPClient, error) {
+	var creds *google.Credentials
+
+	if cfg.STORAGE_EMULATOR_HOST != "" {
+		creds, _ = google.CredentialsFromJSON(ctx, []byte(`{"type": "service_account", "project_id": "my-project-id"}`))
+	} else {
```

Comment:
Is there a project that emulates GCS storage (unauthenticated)? Is this something useful for contributors who want to test this without launching on GCP/GKE?

Reply:
Yes.  There is one storage emulator in python. A couple of storage in go. 
This one works fine: https://github.com/fullstorydev/emulators